prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -34,15 +34,17 @@ class SendFunction {
 
                 val content = workflowEngine.readBody(header)
 
-                val success: Boolean = service.transports.map {
-                    when (it) {
-                        is SFTPTransportType -> SftpTransport().send(service.fullName, it, header, content)
-                        else -> true
+                if (service.transforms.size > 0) {
+                    val success: Boolean = service.transports.map {
+                        when (it) {
+                            is SFTPTransportType -> SftpTransport().send(service.fullName, it, header, content)
+                            else -> true
+                        }
+                    }.reduce { acc, s -> acc && s }
+
+                    if (success) {
+                        context.logger.info("Sent report: ${header.task.reportId} to ${service.fullName}")
                     }
-                }.reduce { acc, s -> acc && s }
```

Comment:
The .map function appears to throw the exception, but also call the following reduce. So, I'm seeing logs that have both an SFTP exception and the map-reduce exception. Here's my non-functional sytle suggestion that avoids the map function.
```
                if (service.transforms.isNotEmpty()) {
                    var success: Boolean = true
                    service.transports.forEach {
                        success = success && when (it) {
                            is SFTPTransportType -> SftpTransport().send(service.fullName, it, header, content)
                            else -> true
                        }
                    }
                    if (success) {
                        context.logger.info("Sent report: ${header.task.reportId} to ${service.fullName}")
                    }
                }
```

Reply:
Inserted...